### ISSUE TITLE
Nginx-DLB with lots of "dlb_send fail with 0 times retry"

### ISSUE THREAD

**Issue by YiXR** [envoyproxy/envoy#26975]
**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via emailing
envoy-security@googlegroups.com where the issue will be triaged appropriately.**

*Title*: *Nginx-DLB with lots of "dlb_send fail with 0 times retry"*

*Description*:
>What issue is being seen? Describe what should be happening instead of
the bug, for example: Envoy should not crash, the expected value isn't
returned, etc.

*Repro steps*:
> Include sample requests, environment, etc. All data and inputs
required to reproduce the bug.

Got envoy contrib v1.26.0. Compiled libdlb according to instructions here (https://downloadmirror.intel.com/727424/DLB_Driver_User_Guide.pdf) and tested libdlb examples to see that they run. Ran envoy with config file shown below, with the DLB snippet copied from this page (https://www.envoyproxy.io/docs/envoy/latest/configuration/other_features/dlb). Got a core dumped error.

>**Note**: The [Envoy_collect tool](https://github.com/envoyproxy/envoy/blob/main/tools/envoy_collect/README.md)
gathers a tarball with debug logs, config and the following admin
endpoints: /stats, /clusters and /server_info. Please note if there are
privacy concerns, sanitize the data prior to sharing the tarball/pasting.

*Admin and Stats Output*:
>Include the admin output for the following endpoints: /stats,
/clusters, /routes, /server_info. For more information, refer to the
[admin endpoint documentation.](https://www.envoyproxy.io/docs/envoy/latest/operations/admin)

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Config*:
>Include the config used to configure Envoy.

```
static_resources:
  listeners:
  - address:
      socket_address:
        address: 0.0.0.0
        port_value: 8081
    connection_balance_config:
      extend_balance:
        name: envoy.network.connection_balance.dlb
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.network.connection_balance.dlb.v3alpha.Dlb
    filter_chains:
    - filters:
      - name: envoy.filters.network.http_connection_manager
        typed_config:
          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
          codec_type: AUTO
          stat_prefix: ingress_http
          route_config:
            name: local_route
            virtual_hosts:
            - name: backend
              domains:
              - "*"
              routes:
              - match:
                  prefix: "/"
                route:
                  cluster: service1
          http_filters:
          - name: envoy.filters.http.router
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router

  clusters:
  - name: service1
    load_assignment:
      cluster_name: service1
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: 127.0.0.1
                port_value: 18081
```
*Logs*:
>Include the access logs and the Envoy logs.

```
[2023-04-26 16:19:54.589][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451242][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_13 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451259][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_29 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451242][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_13 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451242][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_13 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451242][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_13 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451242][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_13 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451242][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_13 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.589][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.590][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.590][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.590][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.590][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.590][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.590][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
[2023-04-26 16:19:54.590][451250][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_20 dlb_send fail with 0 times retry, errono: 11, message: Resource temporarily unavailable
[2023-04-26 16:19:54.590][451222][error][connection] [contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc:266] worker_3 dlb_send fail with 0 times retry, errono: 115, message: Operation now in progress
```

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.

*Call Stack*:
> If the Envoy binary is crashing, a call stack is **required**.
Please refer to the [Bazel Stack trace documentation](https://github.com/envoyproxy/envoy/tree/main/bazel#stack-trace-symbol-resolution).

**Comment by daixiang0** [comment]
The root cause is that the default value is 0, like https://github.com/envoyproxy/envoy/issues/9552

**Comment by adisuissa** [comment]
Fixed by #27009
